util/statvar_dcid_generator.py

```diff
@@ -0,0 +1,139 @@
+        """Generate a name for a quantity.
+        Expected syntax of s- [<value> <quantity>]
+        """
+        str_list = s.split()
+
+        value = str_list[0][1:]
+        quantity = str_list[1][:-1]
+
+        quantity = self.capitalize_word(quantity)
+        return f"{value}{quantity}"
+
+    def get_stat_var_name(self, stat_var_dict, ignore_props=None):
+        """Generates the dcid given a statistical variable
+        
+        Arguments:
+        stat_var_dict -- A dictionary with property:values of the statistical variable as key-value pairs
+        ignore_props -- A list of properties to ignore from stat_var_dict when generating the dcid
+        """
+        dcid = ''
+        has_denominator = False
+        svd = copy.deepcopy(stat_var_dict)
+
+        if ignore_props is not None:
+            for prop in ignore_props:
+                del svd[prop]
+
+        if 'measurementDenominator' in svd:
+            has_denominator = True
+            denominator_dcid = svd['measurementDenominator']
+            del svd['measurementDenominator']
+
+        if 'measurementQualifier' in svd:
+            dcid = self.append_to_dcid(dcid,
+                                       svd['measurementQualifier'],
+                                       end_underscore=True)
+            del svd['measurementQualifier']
+
+        if (svd['statType'] != 'measuredValue'):
+            if svd['statType'] in self.STAT_TYPE_LOOKUP:
+                dcid = self.append_to_dcid(
+                    dcid,
+                    self.STAT_TYPE_LOOKUP[svd['statType']],
+                    end_underscore=True)
+            else:
+                dcid = self.append_to_dcid(dcid,
+                                           svd['statType'],
+                                           end_underscore=True)
+        del svd['statType']
+
+        dcid = self.append_to_dcid(dcid,
+                                   svd['measuredProperty'],
+                                   end_underscore=True)
+        del svd['measuredProperty']
+
+        dcid = self.append_to_dcid(dcid,
+                                   svd['populationType'],
+                                   end_underscore=True)
+        del svd['populationType']
+
+        constraint_props = sorted(svd.keys(), key=str.casefold)
+        for prop in constraint_props:
+            if re.match('\[(-|\d+(\.\d+)?) (-|\d+(\.\d+)?) [a-zA-Z]+\]',
+                        svd[prop]):  #quantity range
+                q_name = self.generate_quantity_range_name(svd[prop])
+                dcid = self.append_to_dcid(dcid, q_name, end_underscore=True)
+
+            elif re.match('\[(-|\d+(\.\d+)?) [a-zA-Z]+\]',
+                          svd[prop]):  #quantity
+                q_name = self.generate_quantity_name(svd[prop])
+                dcid = self.append_to_dcid(dcid, q_name, end_underscore=True)
+
+            else:
+                dcid = self.append_to_dcid(dcid, svd[prop], end_underscore=True)
```

Comment:
Add a TODO for handling edge-cases? Like: https://github.com/datacommonsorg/tools/blob/master/stat_var_renaming/stat_var_renaming.py#L12-L66

Reply:
Done.